### PR TITLE
preprints, adds parsing of pub_history to extract the first preprint

### PR DESCRIPTION
**Pull request by lsh-0**

### CODE REVIEW

**Comment by lsh-0** [comment]
hey @gnott , if you could let me know if you think I've missed anything I'd be grateful.

The changes include some fixing some deprecation warnings and removing some old old commented out code.

File: src/main.py [hunk 1]
```diff
@@ -262,6 +253,22 @@ def body(soup):
 def appendices(soup):
     return jats('appendices_json', base_url(jats('publisher_id')(soup)))(soup)
 
+def preprint_events(struct):
+    "returns a list of 'preprint' type events from article's pub-history or `None`"
+    if not struct or not isinstance(struct, list) or len(struct) == 0:
+        return
+    return [event for event in struct if event.get('type') == 'preprint'] or None
```
Comment (gnott) [inline src/main.py:260]: I think possibly I named this `event_type`, i.e. `if event.get('event_type')`

Comment (lsh-0) [reply src/main.py:260]: it's a good point, I'm not testing against anything coming from elife-tools. Let me double check what elife-tools is returning

Comment (lsh-0) [reply src/main.py:260]: here is your work: https://github.com/elifesciences/issues/issues/6696#issuecomment-850580729-permalink

and it does indeed look like `event_type`, good catch

**Comment by gnott** [comment]
I'm not sure how the output fits into Lax or the API schema, otherwise looking good!

File: src/main.py (hunk 1 above)
Comment (lsh-0) [reply src/main.py:260]: test updated using actual output

**Comment by lsh-0** [comment]
> I'm not sure how the output fits into Lax or the API schema, otherwise looking good!

Regarding the api-raml, articles have a 'article-history' API endpoint that previously was only POA and VOR snippets, but now contain a preprint stub as well. This should only affect applications requesting the latest (or an unspecified) version of the article-history endpoint.

I could have added the preprint information to the header of what the article-history endpoint returns rather than a new item in the list, but I wasn't getting any concrete answers about it's spec. 

With Lax I just need to figure out how to store this preprint information, serve it up and version the endpoint

**Comment by lsh-0** [comment]
test-lax finally made it through testing with that other bot-lax change, merging this in now

**Pull request merged**